core/except.cpp

```diff
@@ -305,5 +305,15 @@ void validExtent(const scipp::index size) {
     throw except::DimensionError("Dimension size cannot be negative.");
 }
 
+void binEdgeOverlaps(const VariableConstProxy &lhs,
+                     const VariableConstProxy &rhs, const Dim dim) {
+  const auto lhsLength = lhs.dims()[dim];
+
+  const auto lastBinOfLhs = lhs.slice({dim, lhsLength - 1, lhsLength});
+  const auto firstBinOfRhs = rhs.slice({dim, 0, 1});
```

Comment:
Is there a reason for slicing with a (size 1) range, instead of just using a single index?